hazelcast-client/src/test/java/com/hazelcast/client/ringbuffer/RingbufferTest.java

```diff
@@ -70,6 +74,13 @@ public void tearDown() {
         hazelcastFactory.terminateAll();
     }
 
+    @Test(expected = ExecutionException.class)
+    public void readManyAsync_whenHitsStale_shouldNotBeBlocked() throws Exception {
+        ICompletableFuture<ReadResultSet<String>> f = clientRingbuffer.readManyAsync(0, 1, 10, null);
+        serverRingbuffer.addAllAsync(asList("0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10"), OVERWRITE);
+        f.get();
```

Comment:
`f.get()` will throw `ExecutionException` wrapping any other kind of exception. Maybe best to match the root cause with an `ExpectedException` rule?
```
expected.expect(new RootCauseMatcher(StaleSequenceException.class));
f.get();
```